### PR TITLE
Don't make post-self-insert-hook local variable.

### PR DESCRIPTION
**Pull request by kurnevsky**
Instead use additional argument to add-hook to add local hooks. This
fixes a case when smartparens is loaded after scala-mode and adds its
hook to post-self-insert-hook which intended to be global.

### CODE REVIEW

**Comment by kurnevsky** [comment]
@hvesalai could you take a look please? :)

File: scala-mode.el [hunk 1]
```diff
@@ -106,7 +106,6 @@ When started, runs `scala-mode-hook'.
 ;  :abbrev
 
   (scala-mode:make-local-variables
-   'post-self-insert-hook
```
Comment (hvesalai) [inline scala-mode.el:109]: why did you remove this?

**Comment by hvesalai** [comment]
I don't understand how post-self-insert-hook could be global? What if I'm coding scala in one buffer and c++ in an other? I wouldn't want the c++ one to have the scala functions.

**Comment by kurnevsky** [comment]
That's why I added additional argument to `add-hook`. All scala functions will be added to scala-mode buffers only.

File: scala-mode-map.el [hunk 2]
```diff
@@ -13,13 +13,13 @@
 
 (defun scala-mode-map:add-self-insert-hooks ()
   (add-hook 'post-self-insert-hook
-            'scala-indent:indent-on-parentheses)
+            'scala-indent:indent-on-parentheses nil t)
```
Comment (hvesalai) [inline scala-mode-map.el:16]: ok, so the `t` means local?

Comment (kurnevsky) [reply scala-mode-map.el:16]: Yes.
```
Signature
(add-hook HOOK FUNCTION &optional DEPTH LOCAL)
```

**Comment by hvesalai** [comment]
Thank you!
